app/scripts/directives/notifications/notificationDrawerWrapper.js

```diff
@@ -0,0 +1,334 @@
+          _.each(group.notifications, function(notification) {
+            notification.unread = false;
+            EventsService.markRead(notification.event);
+          });
+          render();
+          $rootScope.$emit('NotificationDrawerWrapper.onMarkAllRead');
+        },
+        onClearAll: function(group) {
+          _.each(group.notifications, function(notification) {
+            EventsService.markRead(notification.event);
+            EventsService.markCleared(notification.event);
+          });
+          group.notifications = [];
+          render();
+          $rootScope.$emit('NotificationDrawerWrapper.onMarkAllRead');
+        },
+        notificationGroups: notificationGroups,
+        headingInclude: 'views/directives/notifications/header.html',
+        notificationBodyInclude: 'views/directives/notifications/notification-body.html',
+        customScope: {
+          clear: function(notification, index, group) {
+            EventsService.markCleared(notification.event);
+            group.notifications.splice(index, 1);
+            countUnreadNotificationsForAllGroups();
+          },
+          markRead: function(notification) {
+            notification.unread = false;
+            EventsService.markRead(notification.event);
+            countUnreadNotificationsForAllGroups();
+          },
+          getNotficationStatusIconClass: function(event) {
+            return iconClassByEventSeverity[event.type] || iconClassByEventSeverity.info;
+          },
+          getStatusForCount:  function(countKey) {
+            return iconClassByEventSeverity[countKey] || iconClassByEventSeverity.info;
+          },
+          close: function() {
+            drawer.drawerHidden = true;
+          }
+        }
+      });
+
+      var projectChanged = function(next, current) {
+        return _.get(next, 'params.project') !== _.get(current, 'params.project');
+      };
+
+      var reset = function() {
+        getProject($routeParams.project).then(function() {
+          watchEvents($routeParams.project, eventWatchCallback);
+          //watchNotifications($routeParams.project, notificationWatchCallback);
+          hideIfNoProject($routeParams.project);
+          render();
+        });
+      };
+
+      var initWatches = function() {
+        if($routeParams.project) {
+          reset();
+        }
+        // $routeChangeSuccess seems more reliable than $locationChangeSuccess:
+        // - it fires once on initial load. $locationChangeSuccess does not.
+        // - it waits for more object resolution (not a huge deal in this use case)
+        // - tracks route data instead of urls (args to callback fn, also not
+        //   necessary for the current use case)
+        rootScopeWatches.push($rootScope.$on("$routeChangeSuccess", function (evt, next, current) {
+          if(projectChanged(next, current)) {
+            drawer.customScope.projectName = $routeParams.project;
+            reset();
```

Comment:
Can you check that `reset()` is not called twice when first loading the page?

Reply:
Yup, it is only called once.